Flask API: API key leaked

When input an invalid product description, the error message from the API leaked the API key.
```
___________________________________ test_search_fda ___________________________________

    def test_search_fda():
        response = requests.post("https://api.healthly.dev/",
                                 json={"productDescription": "345afdsaf"})
    
        print(response.text)
>       assert response.status_code == 200
E    assert 500 == 200
E     +  where 500 = <Response [500]>.status_code

test_api.py:9: AssertionError
-------------------------------- Captured stdout call ---------------------------------
{"details":"404 Client Error: Not Found for url: https://api.fda.gov/device/enforcement.json?api_key=tq4x******************************************&search=product_description:%22345afdsaf%22&limit=100","error":"Failed to fetch data from the API"}
```

api key is hidden:
```
================================= test session starts =================================
platform linux -- Python 3.8.18, pytest-8.3.3, pluggy-1.5.0 -- /nix/store/dydlfqsd3al5d06h18922sp60b67pk32-python-wrapped-0.1.0/bin/python3
cachedir: .pytest_cache
rootdir: /home/runner/maintest
configfile: pyproject.toml
collected 6 items                                                                     

test_api.py::test_search_fda FAILED                                             [ 16%]
test_api.py::test_search_k510 FAILED                                            [ 33%]
test_api.py::test_search_cdph PASSED                                            [ 50%]
test_api.py::test_search_maude FAILED                                           [ 66%]
test_api.py::test_search_openhistorical FAILED                                  [ 83%]
test_api.py::test_search_cabusinessentity PASSED                                [100%]

====================================== FAILURES =======================================
___________________________________ test_search_fda ___________________________________

    def test_search_fda():
        response = requests.post("https://api.healthly.dev/",
                                 json={"productDescription": "98475982h345hkj"})
    
        print(response.text)
>       assert response.status_code == 200
E    assert 500 == 200
E     +  where 500 = <Response [500]>.status_code

test_api.py:9: AssertionError
-------------------------------- Captured stdout call ---------------------------------
{"details":"404 Client Error: Not Found for url: https://api.fda.gov/device/enforcement.json?api_key=<HIDDEN>&search=product_description:%2298475982h345hkj%22&limit=100","error":"Failed to fetch data from the API"}

__________________________________ test_search_k510 ___________________________________

    def test_search_k510():
        response = requests.post("https://api.healthly.dev/k510",
                                 json={"deviceName": "98475982h345hkj"})
    
        print(response.text)
>       assert response.status_code == 200
E    assert 500 == 200
E     +  where 500 = <Response [500]>.status_code

test_api.py:19: AssertionError
-------------------------------- Captured stdout call ---------------------------------
{"details":"404 Client Error: Not Found for url: https://api.fda.gov/device/510k.json?api_key=<HIDDEN>&search=device_name:%2298475982h345hkj%22&limit=100","error":"Failed to fetch data from the API"}

__________________________________ test_search_maude __________________________________

    def test_search_maude():
        response = requests.post("https://api.healthly.dev/maude",
                                 json={"deviceName": "98475982h345hkj"})
    
        print(response.text)
>       assert response.status_code == 200
E    assert 500 == 200
E     +  where 500 = <Response [500]>.status_code

test_api.py:38: AssertionError
-------------------------------- Captured stdout call ---------------------------------
{"details":"404 Client Error: Not Found for url: https://api.fda.gov/device/event.json?api_key=<HIDDEN>&search=device.generic_name:%2298475982h345hkj%22&limit=100","error":"Failed to fetch data from the API"}

_____________________________ test_search_openhistorical ______________________________

    def test_search_openhistorical():
        response = requests.post("https://api.healthly.dev/openhistorical",
                                 json={"keyword": "98475982h345hkj"})
    
        print(response.text)
>       assert response.status_code == 200
E    assert 500 == 200
E     +  where 500 = <Response [500]>.status_code

test_api.py:48: AssertionError
-------------------------------- Captured stdout call ---------------------------------
{"details":"404 Client Error: Not Found for url: https://api.fda.gov/other/historicaldocument.json?api_key=<HIDDEN>&search=text:%2298475982h345hkj%22&limit=100","error":"Failed to fetch data from the API"}

=============================== short test summary info ===============================
FAILED test_api.py::test_search_fda - assert 500 == 200
 +  where 500 = <Response [500]>.status_code
FAILED test_api.py::test_search_k510 - assert 500 == 200
 +  where 500 = <Response [500]>.status_code
FAILED test_api.py::test_search_maude - assert 500 == 200
 +  where 500 = <Response [500]>.status_code
FAILED test_api.py::test_search_openhistorical - assert 500 == 200
 +  where 500 = <Response [500]>.status_code
============================ 4 failed, 2 passed in 10.09s =============================
```

--- a/AwesomeProject/Backend/main.py
+++ b/AwesomeProject/Backend/main.py
@@ -75,7 +75,7 @@ def search_fda():
         return jsonify(response.json())  # Return the JSON response from the API
     except requests.RequestException as e:
         logging.error(f"Error fetching data from FDA API: {e}")  # Log any errors
-        return jsonify({"error": "Failed to fetch data from the API", "details": str(e)}), 500
+        return jsonify({"error": "Failed to fetch data from the API", "details": str(e).replace(apikey, "<HIDDEN>")}), 500
 
 # Define a new route for K510 database search
 @app.route("/k510", methods=['POST'])
@@ -120,7 +120,7 @@ def search_k510():
         return jsonify(response.json())  # Return the JSON response from the API
     except requests.RequestException as e:
         logging.error(f"Error fetching data from FDA K510 API: {e}")  # Log any errors
-        return jsonify({"error": "Failed to fetch data from the API", "details": str(e)}), 500
+        return jsonify({"error": "Failed to fetch data from the API", "details": str(e).replace(apikey, "<HIDDEN>")}), 500
 
 # Define a new route for CDPH device recall search
 @app.route("/cdph", methods=['POST'])
@@ -213,7 +213,7 @@ def search_maude():
         return jsonify(response.json())  # Return the JSON response from the API
     except requests.RequestException as e:
         logging.error(f"Error fetching data from FDA Maude API: {e}")  # Log any errors
-        return jsonify({"error": "Failed to fetch data from the API", "details": str(e)}), 500
+        return jsonify({"error": "Failed to fetch data from the API", "details": str(e).replace(apikey, "<HIDDEN>")}), 500
 
 # Define a new route for OpenHistorical search
 @app.route("/openhistorical", methods=['POST'])
@@ -329,7 +329,7 @@ def search_openhistorical():
     except requests.RequestException as e:
         logging.error(f"Error fetching data from FDA OpenHistorical API: {e}")
         return (
-            jsonify({"error": "Failed to fetch data from the API", "details": str(e)}),
+            jsonify({"error": "Failed to fetch data from the API", "details": str(e).replace(apikey, "<HIDDEN>")}),
             500,
         )
 

